docs/source/client_upgrades.rst

```diff
@@ -0,0 +1,250 @@
+
+    s3 = session.get_client('s3', 'us-west-2')
+    for obj in s3.list_objects(Bucket='mybucket')['Contents']:
+        name = obj['Key']
+        # Using the same client will reuse any existing HTTP
+        # connections the client was using.
+        print(s3.head_object(Bucket='mybucket', Key=name))
+
+
+Operation and Method Names
+--------------------------
+
+In the old interface, you would retrieve an API operation using the
+casing defined by the service, which is typically CamelCase.  For example,
+you'd use ``service.get_operation('ListObjects')``, not
+``service.get_operation('list_objects')``.  With clients, method names,
+which map 1 - 1 to operation names are snake_cased, as is common in python
+code.
+
+
+**Old**
+
+.. code-block:: python
+
+    service = session.get_service('s3')
+    list_objects = service.get_operation('ListObjects')
+    head_object = service.get_operation('HeadObject')
+    get_object = service.get_operation('GetObject')
+
+**New**
+
+.. code-block:: python
+
+    s3 = session.get_client('s3', 'us-west-2')
+    list_objects = s3.list_objects
+    head_object = s3.head_object
+    get_object = s3.get_object
+
+
+
+Return Values
+-------------
+
+In the old interface, the return value for an ``operation.call`` invocation is
+a tuple of the HTTP response object, and the parsed dict that results from
+parsing the HTTP object.  In the client interface, only the parsed
+response is returned.  The HTTP response object is not returned.
+
+**Old**
+
+.. code-block:: python
+
+    ec2 = session.get_service('ec2')
+    endpoint = ec2.get_endpoint('ec2')
+    describe_instances = ec2.get_operation('DescribeInstances')
+    http, parsed = describe_instances.call(endpoint)
+
+**New**
+
+.. code-block:: python
+
+    ec2 = session.get_client('ec2', 'us-west-2')
+    parsed = ec2.describe_instances()
+
+The main reason for returning the HTTP response was to check if an
+error occurred via the HTTP response status code.  This is now no longer
+required (discussed below).  Exception are automatically raised.
+
+Note that if for some reason you do need to see the response status code,
+it is available via the ``ResponseMetadata`` in the parsed dict that's
+returned.
+
+.. code-block:: python
+
+    ec2 = session.get_client('ec2', 'us-west-2')
+    parsed = ec2.describe_instances()
+    print("The status code is:",
+          parsed['ResponseMetadata']['HTTPStatusCode']))
+
+
+Error Handling
+--------------
+
+With clients, exceptions are now raised on any non 200 response.
+A ``ClientError`` exception has both a ``.msg`` attribute as well
+as the parsed error response (which is a dictionary).  There isn't
+really equivalent functionality in the old interface, but below is
+an example of how you can handle an error:
+
+**New**
+
+.. code-block:: python
+
+    from botocore.exceptions import ClientError
+
+    ec2 = session.get_client('ec2', 'us-west-2')
+    try:
+        parsed = ec2.describe_instances()
```

Comment:
@jamesls 
Can we just make it a call to describe_instances using an instance-id of a non-existent instance?